internal/wrappers/features/documents.feature

```diff
@@ -1,17 +1,210 @@
+    Then the document is successfully created
+
+  Scenario: Replace a document if it exists
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'test-collection'
+    When I createOrReplace a document with id 'my-document-id'
+    Then the document is successfully replaced
+
+  Scenario: Replace a document
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'test-collection'
+    And the collection has a document with id 'replace-my-document-id'
+    When I replace a document with id 'replace-my-document-id'
+    Then the document is successfully replaced
+
+  Scenario: Delete a document
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'test-collection'
+    And the collection has a document with id 'delete-my-document-id'
+    When I delete the document with id 'delete-my-document-id'
+    Then the document is successfully deleted
+
+  Scenario: Update a document
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'test-collection'
+    And the collection has a document with id 'update-my-document-id'
+    When I update a document with id 'update-my-document-id'
+    Then the document is successfully updated
+
+  Scenario: Search a document by id and find it
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'test-collection'
+    And the collection has a document with id 'search-my-document-id'
+    When I search a document with id 'search-my-document-id'
+    Then the document is successfully found
+
+  Scenario: Search a document by id and don't find it
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'test-collection'
+    And the collection has a document with id 'search-my-document-id'
+    When I search a document with id 'fake-id'
+    Then the document is not found
+
+  Scenario: Count documents in a collection
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'count-test-collection'
+    And the collection has a document with id 'count-my-document-id'
+    And the collection has a document with id 'count-my-document-id2'
+    And the collection has a document with id 'count-my-document-id3'
+    When I count how many documents there is in the collection
+    Then I shall receive 3
+
+  Scenario: Delete multiple documents with no error
+    Given Kuzzle Server is running
+    And there is an index 'test-index'
+    And it has a collection 'delete-test-collection'
+    And the collection has a document with id 'mdelete-my-document-id'
+    And the collection has a document with id 'mdelete-my-document-id2'
+    When I delete the documents 'mdelete-my-document-id' and 'mdelete-my-document-id2'
+    And I count how many documents there is in the collection
```

Comment:
This would be the When step:
When I count how many documents there is in the collection

Reply:
Why? I am testing the delete feature there so I say when I delete a document and I count then I should have <number> documents

Reply:
Because in the scenario, the step that allow you to check everything is working, is when you count the documents in the collection, and check that the number of documents is coherent with the expected number, considering the previous steps